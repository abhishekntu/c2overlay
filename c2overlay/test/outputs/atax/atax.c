/****************************************************************/
/*   chebyshev		                                        */
/****************************************************************/

#ifdef TEST
#include <stdio.h>
#endif

#include<math.h>

#ifdef ALGO
void foo()
{
    int A [9];
    int x [3];
    int y [3];
    int i;
    {
        y[0] = (A[0] * (A[0] * x[0] + A[1] * x[1] + A[2] * x[2]) + A[3] * (A[3] * x[0] + A[4] * x[1] + A[5] * x[2]) + A[6] * (A[6] * x[0] + A[7] * x[1] + A[8] * x[2]));
        y[1] = (A[1] * (A[0] * x[0] + A[1] * x[1] + A[2] * x[2]) + A[4] * (A[3] * x[0] + A[4] * x[1] + A[5] * x[2]) + A[7] * (A[6] * x[0] + A[7] * x[1] + A[8] * x[2]));
        y[2] = (A[2] * (A[0] * x[0] + A[1] * x[1] + A[2] * x[2]) + A[5] * (A[3] * x[0] + A[4] * x[1] + A[5] * x[2]) + A[8] * (A[6] * x[0] + A[7] * x[1] + A[8] * x[2]));
    }

}
#endif


#ifdef TEST
int main(void)
{
	int x;
 	int result;
  	foo();
  	return 0;
}
#endif
