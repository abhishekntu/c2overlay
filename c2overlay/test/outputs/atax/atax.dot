digraph graphname {
N49 [color="black", label="store_Imm_1_N74_N49"];
N48 [color="black", label="N37_mul_N37_N48"];
N41 [color="black", label="N50_mul_N50_N41"];
N40 [color="black", label="N53N61N54_add_N53_mul_N61_add_N54_N40"];
N43 [color="black", label="N39_mul_N39_N43"];
N42 [color="black", label="N51_mul_N51_N42"];
N45 [color="black", label="N33N13_mul_N33_add_N13_N45"];
N44 [color="black", label="N32_mul_N32_N44"];
N47 [color="black", label="N35_mul_N35_N47"];
N46 [color="black", label="N34N31_mul_N34_add_N31_N46"];
N29 [color="black", label="N23N38_mul_N23_add_N38_N29"];
N28 [color="black", label="N59_mul_N59_N28"];
N23 [color="black", label="N45N65N16_add_N45_mul_N65_add_N16_N23"];
N22 [color="black", label="N42N56_mul_N42_add_N56_N22"];
N21 [color="black", label="N43_mul_N43_N21"];
N20 [color="black", label="N41N60_mul_N41_add_N60_N20"];
N27 [color="black", label="N14N30N15_add_N14_mul_N30_add_N15_N27"];
N26 [color="black", label="N29_mul_N29_N26"];
N25 [color="black", label="N69N57N68_add_N69_mul_N57_add_N68_N25"];
N24 [color="black", label="N44_mul_N44_N24"];
N8 [color="black", label="load_Imm_3_N5_N8"];
N9 [color="black", label="load_Imm_1_N6_N9"];
N1 [color="black", label="load_Imm_1_N12_N1"];
N2 [color="black", label="load_Imm_0_N8_N2"];
N3 [color="black", label="load_Imm_0_N9_N3"];
N4 [color="black", label="load_Imm_8_N1_N4"];
N5 [color="black", label="load_Imm_5_N2_N5"];
N6 [color="black", label="load_Imm_6_N3_N6"];
N7 [color="black", label="load_Imm_4_N4_N7"];
N12 [color="black", label="load_Imm_2_N11_N12"];
N13 [color="black", label="N55N70_mul_N55_add_N70_N13"];
N10 [color="black", label="load_Imm_7_N7_N10"];
N11 [color="black", label="load_Imm_2_N10_N11"];
N16 [color="black", label="N67N36_mul_N67_add_N36_N16"];
N17 [color="black", label="N66N47_mul_N66_add_N47_N17"];
N14 [color="black", label="N46N27_mul_N46_add_N27_N14"];
N15 [color="black", label="N19N72N18_add_N19_mul_N72_add_N18_N15"];
N18 [color="black", label="N64_mul_N64_N18"];
N19 [color="black", label="N63N17N40_add_N63_mul_N17_add_N40_N19"];
N50 [color="black", label="store_Imm_2_N75_N50"];
N51 [color="black", label="store_Imm_0_N73_N51"];
N38 [color="black", label="N58_mul_N58_N38"];
N39 [color="black", label="N52N62_mul_N52_add_N62_N39"];
N30 [color="black", label="N22_mul_N22_N30"];
N31 [color="black", label="N21N20_mul_N21_add_N20_N31"];
N32 [color="black", label="N26_mul_N26_N32"];
N33 [color="black", label="N25_mul_N25_N33"];
N34 [color="black", label="N24_mul_N24_N34"];
N35 [color="black", label="N49N28_mul_N49_add_N28_N35"];
N36 [color="black", label="N48_mul_N48_N36"];
N37 [color="black", label="N71_mul_N71_N37"];
N48 -> N45;
N41 -> N13;
N40 -> N19;
N43 -> N40;
N42 -> N39;
N45 -> N27;
N44 -> N14;
N47 -> N27;
N46 -> N47;
N29 -> N23;
N28 -> N40;
N23 -> N25;
N22 -> N40;
N21 -> N20;
N20 -> N46;
N27 -> N15;
N26 -> N27;
N25 -> N50;
N24 -> N22;
N8 -> N17;
N8 -> N22;
N8 -> N45;
N8 -> N40;
N9 -> N20;
N9 -> N44;
N9 -> N47;
N9 -> N29;
N1 -> N44;
N1 -> N29;
N1 -> N48;
N1 -> N31;
N1 -> N13;
N1 -> N33;
N1 -> N20;
N1 -> N42;
N1 -> N24;
N2 -> N34;
N2 -> N43;
N2 -> N14;
N2 -> N21;
N3 -> N41;
N3 -> N32;
N3 -> N14;
N3 -> N39;
N3 -> N21;
N3 -> N17;
N3 -> N45;
N3 -> N22;
N3 -> N34;
N4 -> N38;
N4 -> N37;
N4 -> N30;
N4 -> N25;
N5 -> N36;
N5 -> N28;
N5 -> N26;
N5 -> N35;
N6 -> N41;
N6 -> N39;
N6 -> N32;
N6 -> N19;
N7 -> N33;
N7 -> N24;
N7 -> N48;
N7 -> N27;
N12 -> N30;
N12 -> N35;
N12 -> N28;
N12 -> N16;
N12 -> N37;
N12 -> N18;
N12 -> N26;
N12 -> N38;
N12 -> N46;
N13 -> N25;
N10 -> N13;
N10 -> N31;
N10 -> N42;
N10 -> N15;
N11 -> N16;
N11 -> N46;
N11 -> N18;
N11 -> N23;
N16 -> N43;
N17 -> N35;
N14 -> N16;
N15 -> N49;
N18 -> N23;
N19 -> N51;
N38 -> N25;
N39 -> N19;
N30 -> N15;
N31 -> N15;
N32 -> N31;
N33 -> N17;
N34 -> N29;
N35 -> N36;
N36 -> N23;
N37 -> N19;
}
