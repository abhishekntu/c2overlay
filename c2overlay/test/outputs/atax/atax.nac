procedure foo()
{
  localvar s32 D_2406;
  localvar s32 D_2407;
  localvar s32 D_2408;
  localvar s32 D_2409;
  localvar s32 D_2410;
  localvar s32 D_2411;
  localvar s32 D_2412;
  localvar s32 D_2413;
  localvar s32 D_2414;
  localvar s32 D_2415;
  localvar s32 D_2416;
  localvar s32 D_2417;
  localvar s32 D_2418;
  localvar s32 D_2419;
  localvar s32 D_2420;
  localvar s32 D_2421;
  localvar s32 D_2422;
  localvar s32 D_2423;
  localvar s32 D_2424;
  localvar s32 D_2425;
  localvar s32 D_2426;
  localvar s32 D_2427;
  localvar s32 D_2428;
  localvar s32 D_2429;
  localvar s32 D_2430;
  localvar s32 D_2431;
  localvar s32 D_2432;
  localvar s32 D_2433;
  localvar s32 D_2434;
  localvar s32 D_2435;
  localvar s32 D_2436;
  localvar s32 D_2437;
  localvar s32 D_2438;
  localvar s32 D_2439;
  localvar s32 D_2440;
  localvar s32 D_2441;
  localvar s32 D_2442;
  localvar s32 D_2443;
  localvar s32 D_2444;
  localvar s32 D_2445;
  localvar s32 D_2446;
  localvar s32 D_2447;
  localvar s32 A[9];
  localvar s32 x[3];
  localvar s32 y[3];
  localvar s32 i;
  localvar s32 t00000;
  localvar s32 t00001;
  localvar s32 t00002;
  localvar s32 t00003;
  localvar s32 t00004;
  localvar s32 t00005;
  localvar s32 t00006;
  localvar s32 t00007;
  localvar s32 t00008;
  localvar s32 t00009;
  localvar s32 t00010;
  localvar s32 t00011;
  localvar s32 t00012;
  localvar s32 t00013;
  localvar s32 t00014;
  localvar s32 t00015;
  localvar s32 t00016;
  localvar s32 t00017;
  localvar s32 t00018;
  localvar s32 t00019;
  localvar s32 t00020;
  localvar s32 t00021;
  localvar s32 t00022;
  localvar s32 t00023;
  localvar s32 t00024;
  localvar s32 t00025;
  localvar s32 t00026;
  localvar s32 t00027;
  localvar s32 t00028;
  localvar s32 t00029;
  localvar s32 t00030;
  localvar s32 t00031;
  localvar s32 t00032;
  localvar s32 t00033;
  localvar s32 t00034;
  localvar s32 t00035;
  localvar s32 t00036;
  localvar s32 t00037;
  localvar s32 t00038;
  localvar s32 t00039;
  localvar s32 t00040;
  localvar s32 t00041;
  localvar s32 t00042;
  localvar s32 t00043;
  localvar s32 t00044;
  localvar s32 t00045;
  localvar s32 t00046;
  localvar s32 t00047;
  localvar s32 t00048;
  localvar s32 t00049;
  localvar s32 t00050;
  localvar s32 t00051;
  localvar s32 t00052;
  localvar s32 t00053;
  localvar s32 t00054;
  localvar s32 t00055;
  localvar s32 t00056;
  localvar s32 t00057;
  localvar s32 t00058;
  localvar s32 t00059;
  localvar s32 t00060;
  localvar s32 t00061;
  localvar s32 t00062;
  localvar s32 t00063;
  localvar s32 t00064;
  localvar s32 t00065;
L0005:
  D_2406 <= load A,0;
  D_2406 <= mov D_2406;
  D_2407 <= load x,0;
  D_2408 <= mul D_2406,D_2407;
  D_2409 <= load A,1;
  D_2410 <= load x,1;
  D_2411 <= mul D_2409,D_2410;
  D_2412 <= add D_2408,D_2411;
  D_2413 <= load A,2;
  D_2414 <= load x,2;
  D_2415 <= mul D_2413,D_2414;
  D_2416 <= add D_2412,D_2415;
  D_2417 <= mul D_2406,D_2416;
  D_2418 <= load A,3;
  D_2418 <= mov D_2418;
  D_2407 <= load x,0;
  D_2419 <= mul D_2418,D_2407;
  D_2420 <= load A,4;
  D_2410 <= load x,1;
  D_2421 <= mul D_2420,D_2410;
  D_2422 <= add D_2419,D_2421;
  D_2423 <= load A,5;
  D_2414 <= load x,2;
  D_2424 <= mul D_2423,D_2414;
  D_2425 <= add D_2422,D_2424;
  D_2426 <= mul D_2418,D_2425;
  D_2427 <= add D_2417,D_2426;
  D_2428 <= load A,6;
  D_2428 <= mov D_2428;
  D_2407 <= load x,0;
  D_2429 <= mul D_2428,D_2407;
  D_2430 <= load A,7;
  D_2410 <= load x,1;
  D_2431 <= mul D_2430,D_2410;
  D_2432 <= add D_2429,D_2431;
  D_2433 <= load A,8;
  D_2414 <= load x,2;
  D_2434 <= mul D_2433,D_2414;
  D_2435 <= add D_2432,D_2434;
  D_2436 <= mul D_2428,D_2435;
  D_2437 <= add D_2427,D_2436;
  y <= store D_2437,0;
  D_2409 <= load A,1;
  D_2406 <= load A,0;
  D_2407 <= load x,0;
  D_2408 <= mul D_2406,D_2407;
  D_2409 <= load A,1;
  D_2410 <= load x,1;
  D_2411 <= mul D_2409,D_2410;
  D_2412 <= add D_2408,D_2411;
  D_2413 <= load A,2;
  D_2414 <= load x,2;
  D_2415 <= mul D_2413,D_2414;
  D_2416 <= add D_2412,D_2415;
  D_2438 <= mul D_2409,D_2416;
  D_2420 <= load A,4;
  D_2418 <= load A,3;
  D_2407 <= load x,0;
  D_2419 <= mul D_2418,D_2407;
  D_2420 <= load A,4;
  D_2410 <= load x,1;
  D_2421 <= mul D_2420,D_2410;
  D_2422 <= add D_2419,D_2421;
  D_2423 <= load A,5;
  D_2414 <= load x,2;
  D_2424 <= mul D_2423,D_2414;
  D_2425 <= add D_2422,D_2424;
  D_2439 <= mul D_2420,D_2425;
  D_2440 <= add D_2438,D_2439;
  D_2430 <= load A,7;
  D_2428 <= load A,6;
  D_2407 <= load x,0;
  D_2429 <= mul D_2428,D_2407;
  D_2430 <= load A,7;
  D_2410 <= load x,1;
  D_2431 <= mul D_2430,D_2410;
  D_2432 <= add D_2429,D_2431;
  D_2433 <= load A,8;
  D_2414 <= load x,2;
  D_2434 <= mul D_2433,D_2414;
  D_2435 <= add D_2432,D_2434;
  D_2441 <= mul D_2430,D_2435;
  D_2442 <= add D_2440,D_2441;
  y <= store D_2442,1;
  D_2413 <= load A,2;
  D_2406 <= load A,0;
  D_2407 <= load x,0;
  D_2408 <= mul D_2406,D_2407;
  D_2409 <= load A,1;
  D_2410 <= load x,1;
  D_2411 <= mul D_2409,D_2410;
  D_2412 <= add D_2408,D_2411;
  D_2413 <= load A,2;
  D_2414 <= load x,2;
  D_2415 <= mul D_2413,D_2414;
  D_2416 <= add D_2412,D_2415;
  D_2443 <= mul D_2413,D_2416;
  D_2423 <= load A,5;
  D_2418 <= load A,3;
  D_2407 <= load x,0;
  D_2419 <= mul D_2418,D_2407;
  D_2420 <= load A,4;
  D_2410 <= load x,1;
  D_2421 <= mul D_2420,D_2410;
  D_2422 <= add D_2419,D_2421;
  D_2423 <= load A,5;
  D_2414 <= load x,2;
  D_2424 <= mul D_2423,D_2414;
  D_2425 <= add D_2422,D_2424;
  D_2444 <= mul D_2423,D_2425;
  D_2445 <= add D_2443,D_2444;
  D_2433 <= load A,8;
  D_2428 <= load A,6;
  D_2407 <= load x,0;
  D_2429 <= mul D_2428,D_2407;
  D_2430 <= load A,7;
  D_2410 <= load x,1;
  D_2431 <= mul D_2430,D_2410;
  D_2432 <= add D_2429,D_2431;
  D_2433 <= load A,8;
  D_2414 <= load x,2;
  D_2434 <= mul D_2433,D_2414;
  D_2435 <= add D_2432,D_2434;
  D_2446 <= mul D_2433,D_2435;
  D_2447 <= add D_2445,D_2446;
  y <= store D_2447,2;
  L_EXIT <= jmpun;
L_EXIT:
  nop;
}
