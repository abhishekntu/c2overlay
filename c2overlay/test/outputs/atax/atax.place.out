Netlist file: test.net   Architecture file: test.arch.xml
Array size: 7 x 7 logic blocks

#block name	x	y	subblk	block number
#----------	--	--	------	------------
N8		4	0	0	#0
N9		0	1	0	#1
N1		2	0	0	#2
N2		3	0	0	#3
N3		6	0	0	#4
N4		8	4	0	#5
N5		0	5	0	#6
N6		8	2	0	#7
N7		4	8	0	#8
N12		0	4	0	#9
N10		7	0	0	#10
N11		0	3	0	#11
out:N49		8	3	0	#12
out:N50		8	5	0	#13
out:N51		5	0	0	#14
N48_blk		5	4	0	#15
N41_blk		6	3	0	#16
N40_blk		4	2	0	#17
N43_blk		3	2	0	#18
N42_blk		7	1	0	#19
N45_blk		4	4	0	#20
N44_blk		2	1	0	#21
N47_blk		1	3	0	#22
N46_blk		1	2	0	#23
N29_blk		2	3	0	#24
N28_blk		1	5	0	#25
N23_blk		2	4	0	#26
N22_blk		5	2	0	#27
N21_blk		4	1	0	#28
N20_blk		1	1	0	#29
N27_blk		4	3	0	#30
N26_blk		3	4	0	#31
N25_blk		7	5	0	#32
N24_blk		5	3	0	#33
N13_blk		7	4	0	#34
N16_blk		2	2	0	#35
N17_blk		4	5	0	#36
N14_blk		3	1	0	#37
N15_blk		7	3	0	#38
N18_blk		1	4	0	#39
N19_blk		5	1	0	#40
N38_blk		6	5	0	#41
N39_blk		6	1	0	#42
N30_blk		6	4	0	#43
N31_blk		7	2	0	#44
N32_blk		6	2	0	#45
N33_blk		4	6	0	#46
N34_blk		3	3	0	#47
N35_blk		3	5	0	#48
N36_blk		2	5	0	#49
N37_blk		5	5	0	#50
